200075/7465.py

```diff
@@ -0,0 +1,34 @@
+T = int(input())
+for t in range(1,T+1) :
+    N, M = map(int, input().split())
+    list1 = [[] for i in range(N + 1)]
+
+    for i in range(M) :
+        n1, n2 = map(int, input().split())
+        list1[n1].append(n2)
+        list1[n2].append(n1)
+
+    # print(list1)
+    visited = [False for i in range(N + 1)]
+    # print(visited)
+
+    cnt = 0
+    while False in visited :
+        for i in range(len(visited)) :
+            # print(i)
+            if visited[i] == False :
+                visited[i] = True
+                start = i
+                break
+        stack = [start]
+        while stack :
+            # print(f'visited = {visited}')
+            for i in list1[stack.pop()] :
+                # print(f'i = {i}')  
+                if visited[i] == False :
+                    visited[i] = True
+                    stack.append(i)
+                # print(visited)
+                # print(list1)  
+        cnt +=1
+    print(f'#{t} {cnt-1}')
```

Comment:
현재 코드는 0 ~ n -1 번의 마을을 탐색합니다! 0번 마을은 없어서 `cnt` 값이 1 더 크게 나왔던 거 같네요.
```suggestion
    cnt = 0
    for i in range(1, N + 1):
        if not visited[i]:          # i번째 마을을 아직 방문 안 했다면
            cnt += 1                # 답을 1 늘리고
            stack = [i]             # dfs 시작!
            while stack:
                for j in list1[stack.pop()]:
                    if not visited[i]:
                        visited[i] = True
                        stack.append(i)
    print(f'#{t} {cnt}')
```
코드를 약간 다듬어봤습니다! 😄  
1. 1번부터 n번 마을을 차례대로 보는데,
2. 아직 방문하지 않은 마을이 나오면 답을 1 늘리고
3. 그 마을에서 dfs 탐색 시작. 탐색하면서 방문한 마을 표시해주기
4. 답 출력